front/app/components/UI/Checkbox/index.tsx

```diff
@@ -104,18 +105,25 @@ export default class Checkbox extends PureComponent<Props> {
       indeterminate,
       onChange,
       name,
+      checkBoxTabIndex,
     } = this.props;
     const hasLabel = !!label;
 
     if (size && isBoolean(disabled) && isBoolean(indeterminate)) {
       return (
-        <Label id={id} className={className || ''} disabled={disabled}>
+        <Label
+          id={id}
+          className={className || ''}
+          disabled={disabled}
+          role="checkbox"
+          aria-checked={checked}
+        >
           <CheckboxContainer hasLabel={hasLabel}>
             <HiddenCheckbox
               onChange={onChange}
               checked={checked}
               disabled={disabled}
-              tabIndex={0}
+              tabIndex={checkBoxTabIndex}
```

Comment:
Assuming you are not putting 0 as the default value because that is the default value of tabIndex anyway?